torch/_inductor/compile_fx.py

```diff
@@ -307,14 +307,30 @@ def compile_fx_inner(
             if isinstance(t, torch.Tensor)
         )
 
-        if (
-            set(graph.device_types) == {"cuda"}
-            and not graph.mutated_inputs
-            and not has_incompatible_cudagraph_ops(gm)
-            and not complex_memory_overlap_inputs
-            and all(isinstance(t, torch.Tensor) for t in example_inputs)
-            and (len(graph.device_idxs) == 1 or not config.triton.cudagraph_trees)
-        ):
+        cudagraph_tests = [
+            (set(graph.device_types) == {"cuda"}, "non-cuda device in graph"),
+            (not graph.mutated_inputs, "mutated inputs"),
+            (not has_incompatible_cudagraph_ops(gm), "incompatible ops"),
+            (not complex_memory_overlap_inputs, "complex memory overlap"),
+            (
+                all(
+                    isinstance(t, (torch.Tensor, torch.SymInt)) for t in example_inputs
+                ),
+                "non-Tensor inputs",
+            ),
```

Comment:
Also do we need to worry about non-Tensor outputs ?

Reply:
Hmm, good question. SymInt outputs are permissible with dynamic shapes. Today, these would be static if all the inputs were static. So it seems in principle it should work.

Reply:
I don't see how you can assert this? You would have to run the original graph to get what the true sizes are, but that would defeat the purpose of cudagraphs??